### ISSUE TITLE
aosx_1: the guest initialization hit the critical timeout, analysis aborted.

### ISSUE THREAD

**Issue by lovina37** [cuckoosandbox/cuckoo#1281]
lovina@ubuntu:~/cuckoo$ python cuckoo.py

   ______   __  __   ______   ___   ___   ______   ______      
  /_____/\ /_/\/_/\ /_____/\ /___/\/__/\ /_____/\ /_____/\     
  \:::__\/ \:\ \:\ \\:::__\/ \::.\ \\ \ \\:::_ \ \\:::_ \ \    
   \:\ \  __\:\ \:\ \\:\ \  __\:: \/_) \ \\:\ \ \ \\:\ \ \ \   
    \:\ \/_/\\:\ \:\ \\:\ \/_/\\:. __  ( ( \:\ \ \ \\:\ \ \ \  
     \:\_\ \ \\:\_\:\ \\:\_\ \ \\: \ )  \ \ \:\_\ \ \\:\_\ \ \ 
      \_____\/ \_____\/ \_____\/ \__\/\__\/  \_____\/ \_____\/

 Cuckoo Sandbox 1.2
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 Outdated! Cuckoo Sandbox version 2.0-rc1 is available now.

2017-02-10 12:14:42,723 [root] INFO: Generating grammar tables from /usr/lib/python2.7/lib2to3/Grammar.txt
2017-02-10 12:14:42,757 [root] INFO: Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt
/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/compiler.py:572: SAWarning: Can't resolve label reference 'added_on desc'; converting to text() (this warning may be suppressed after 10 occurrences)
  util.ellipses_string(element.element))
2017-02-10 12:14:43,114 [lib.cuckoo.core.scheduler] INFO: Using "avd" machine manager
2017-02-10 12:14:43,194 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2017-02-10 12:14:43,209 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.
/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/compiler.py:572: SAWarning: Can't resolve label reference 'priority desc, added_on'; converting to text() (this warning may be suppressed after 10 occurrences)
  util.ellipses_string(element.element))
2017-02-10 12:17:05,974 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "/tmp/cuckoo-tmp/upload_Ms40VK/locker-6ca7420a4c46a627b20d95ff8f54d9b7c78df08efdb1dca17168f14c60cb0a68.apk" (task=5)
2017-02-10 12:17:06,069 [lib.cuckoo.core.scheduler] INFO: Task #5: acquired machine aosx_1 (label=aosx_1)
2017-02-10 12:17:06,108 [modules.auxiliary.sniffer] INFO: Started sniffer with PID 6880 (interface=vboxnet0, host=127.0.0.1, dump path=/home/lovina/cuckoo/storage/analyses/5/dump.pcap)
2017-02-10 12:18:28,835 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=aosx_1, ip=127.0.0.1)
2017-02-10 12:21:29,747 [lib.cuckoo.core.scheduler] ERROR: aosx_1: the guest initialization hit the critical timeout, analysis aborted.
2017-02-10 12:21:29,815 [modules.machinery.avd] INFO: Stopping AVD listening on port 5554
2017-02-10 12:21:31,331 [modules.processing.apkinfo] WARNING: Dex size bigger than: None
2017-02-10 12:21:31,335 [modules.processing.behavior] WARNING: Analysis results folder does not exist at path "/home/lovina/cuckoo/storage/analyses/5/logs".
2017-02-10 12:21:31,534 [lib.cuckoo.common.objects] WARNING: Unable to import pydeep (install with `pip install pydeep`)
2017-02-10 12:21:31,535 [lib.cuckoo.common.objects] WARNING: Unable to import yara (please compile from sources)
2017-02-10 12:21:32,519 [lib.cuckoo.core.plugins] ERROR: Failed to run the reporting module "MongoDB":
Traceback (most recent call last):
  File "/home/lovina/cuckoo/lib/cuckoo/core/plugins.py", line 505, in process
    current.run(self.results)
  File "/home/lovina/cuckoo/modules/reporting/mongodb.py", line 216, in run
    self.conn.disconnect()
  File "/usr/local/lib/python2.7/dist-packages/pymongo/database.py", line 1053, in __call__
    self.__name, self.__client.__class__.__name__))
TypeError: 'Database' object is not callable. If you meant to call the 'disconnect' method on a 'MongoClient' object it is failing because no such method exists.
2017-02-10 12:21:32,774 [lib.cuckoo.core.plugins] WARNING: The reporting module "ReportAndroidHTML" returned the following error: Failed to generate HTML report: 'dict object' has no attribute 'debug'
2017-02-10 12:21:32,937 [lib.cuckoo.core.plugins] WARNING: The reporting module "ReportHTML" returned the following error: Failed to generate HTML report: 'dict object' has no attribute 'debug'
2017-02-10 12:21:32,938 [lib.cuckoo.core.scheduler] INFO: Task #5: reports generation completed (path=/home/lovina/cuckoo/storage/analyses/5)
2017-02-10 12:21:32,970 [lib.cuckoo.core.scheduler] INFO: Task #5: analysis procedure completed

^Clovina@ubuntu:~/cuckoo$ mongo cuckoo
MongoDB shell version: 2.4.9
connecting to: cuckoo
Welcome to the MongoDB shell.
For interactive help, type "help".
For more comprehensive documentation, see
	http://docs.mongodb.org/
Questions? Try the support group
	http://groups.google.com/group/mongodb-user
> ^C
bye

**Comment by lovina37** [comment]
Please tell me where am I going wrong ?? I have tried a lot still its not giving me dynamic analysis ?? It does start the android emulator ... when everything is closed and I start the analysis It open the android emulator ... but then nothing happens .... Is it because I am analyzing android ransomware apk ??

**Comment by doomedraven** [comment]
first of all why 1.2 if already 2rc2 available,  and you have a lot of dependencies missed

**Comment by lovina37** [comment]
How do I update to 2rc2 and how to install the missing dependencies.... is there any place or document where I can have the exact and simplified steps to get cucook droid working on ubuntu

**Comment by lovina37** [comment]
Code  Issues 208  Pull requests 39  Projects 0  Wiki  Pulse  Graphs
￼Edit New issue
aosx_1: the guest initialization hit the critical timeout, analysis aborted. #1281
 Open	lovina37 opened this issue 9 hours ago · 3 comments

**Comment by lovina37** [comment]
lovina@ubuntu:~/cuckoo$ python cuckoo.py

                               ),-.     /
  Cuckoo Sandbox              <(a  `---',' 
     no chance for malwares!  ( `-, ._> )
                               ) _>.___/
                                   _/

 Cuckoo Sandbox 2.0-dev
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 Outdated! Cuckoo Sandbox version 2.0-rc1 is available now.
2017-02-10 21:32:17,627 [lib.cuckoo.core.resultserver] WARNING: Cannot bind ResultServer on port 2042, trying another port.
2017-02-10 21:32:17,630 [lib.cuckoo.core.scheduler] INFO: Using "avd" as machine manager
2017-02-10 21:32:17,935 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2017-02-10 21:32:17,978 [lib.cuckoo.core.scheduler] INFO: Unable to determine the network interface for VM with name aosx_1, Cuckoo will not be able to give it full internet access or route it through a VPN! Please define a default network interface for the machinery or define a network interface for each VM.
2017-02-10 21:32:17,979 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.
2017-02-10 21:34:45,039 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "locker-d429e31f30bc4f13db80b4ec0f9195f623fd7303a9ba52b4537b92d75d80c63b.apk" (task #1, options "")
2017-02-10 21:34:45,103 [lib.cuckoo.core.scheduler] INFO: Task #1: acquired machine aosx_1 (label=aosx_1)
2017-02-10 21:34:45,104 [modules.auxiliary.sniffer] ERROR: Network interface not defined, network capture aborted
2017-02-10 21:34:45,107 [lib.cuckoo.common.objects] WARNING: Error extracting package and main activity: unpack requires a string argument of length 4.
2017-02-10 21:34:45,150 [modules.machinery.avd] INFO: Stopping AVD listening on port 5554
2017-02-10 21:34:46,521 [lib.cuckoo.core.scheduler] ERROR: Failure in AnalysisManager.run
Traceback (most recent call last):
  File "/home/lovina/cuckoo/lib/cuckoo/core/scheduler.py", line 447, in run
    self.launch_analysis()
  File "/home/lovina/cuckoo/lib/cuckoo/core/scheduler.py", line 325, in launch_analysis
    machinery.start(self.machine.label, self.task)
TypeError: start() takes exactly 2 arguments (3 given)

**Comment by lovina37** [comment]
now I get this errors

**Comment by doomedraven** [comment]
that is rc1, update to rc2
http://docs.cuckoosandbox.org/en/latest/

**Comment by lovina37** [comment]
__  __   ______   ___   ___   ______   ______      
  /_____/\ /_/\/_/\ /_____/\ /___/\/__/\ /_____/\ /_____/\     
  \:::__\/ \:\ \:\ \\:::__\/ \::.\ \\ \ \\:::_ \ \\:::_ \ \    
   \:\ \  __\:\ \:\ \\:\ \  __\:: \/_) \ \\:\ \ \ \\:\ \ \ \   
    \:\ \/_/\\:\ \:\ \\:\ \/_/\\:. __  ( ( \:\ \ \ \\:\ \ \ \  
     \:\_\ \ \\:\_\:\ \\:\_\ \ \\: \ )  \ \ \:\_\ \ \\:\_\ \ \ 
      \_____\/ \_____\/ \_____\/ \__\/\__\/  \_____\/ \_____\/

 Cuckoo Sandbox 2.0-dev
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 Outdated! Cuckoo Sandbox version 2.0-rc1 is available now.
2017-02-11 18:32:23,737 [lib.cuckoo.core.scheduler] INFO: Using "avd" as machine manager
2017-02-11 18:32:23,871 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2017-02-11 18:32:23,993 [lib.cuckoo.core.scheduler] INFO: **Unable to determine the network interface for VM with name aosx_1, Cuckoo will not be able to give it full internet access or route it through a VPN! Please define a default network interface for the machinery or define a network interface for each VM.**
2017-02-11 18:32:23,993 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.


http://hydrasky.com/malware-analysis/how-to-install-cuckoodroid/

I am trying to follow this link  to get cuckoo droid running ... sO WHILE cloning I did use git clone --depth=1 https://github.com/cuckoobox/cuckoo.git cuckoo -b 2.0-rc2 still its the same version is it coz I alter the configuration file especially the auxiliary the conf file.

**Comment by lovina37** [comment]
**[cuckoo]**
# Enable or disable startup version check. When enabled, Cuckoo will connect
# to a remote location to verify whether the running version is the latest
# one available.
version_check = on

# If turned on, Cuckoo will delete the original file after its analysis
# has been completed.
delete_original = off

# If turned on, Cuckoo will delete the copy of the original file in the
# local binaries repository after the analysis has finished. (On *nix this
# will also invalidate the file called "binary" in each analysis directory,
# as this is a symlink.)
delete_bin_copy = off

# Specify the name of the machinery module to use, this module will
# define the interaction between Cuckoo and your virtualization software
# of choice.
machinery = avd

# Enable creation of memory dump of the analysis machine before shutting
# down. Even if turned off, this functionality can also be enabled at
# submission. Currently available for: VirtualBox and libvirt modules (KVM).
memory_dump = off

# When the timeout of an analysis is hit, the VM is just killed by default.
# For some long-running setups it might be interesting to terminate the
# monitored processes before killing the VM so that connections are closed.
terminate_processes = off

# Enable automatically re-schedule of "broken" tasks each startup.
# Each task found in status "processing" is re-queued for analysis.
reschedule = off

# Enable processing of results within the main cuckoo process.
# This is the default behavior but can be switched off for setups that
# require high stability and process the results in a separate task.
process_results = on

# Limit the amount of analysis jobs a Cuckoo process goes through.
# This can be used together with a watchdog to mitigate risk of memory leaks.
max_analysis_count = 0

# Limit the number of concurrently executing analysis machines.
# This may be useful on systems with limited resources.
# Set to 0 to disable any limits.
max_machines_count = 0

# Limit the amount of VMs that are allowed to start in parallel. Generally
# speaking starting the VMs is one of the more CPU intensive parts of the
# actual analysis. This option tries to avoid maxing out the CPU completely.
max_vmstartup_count = 10

# Minimum amount of free space (in MB) available before starting a new task. 
# This tries to avoid failing an analysis because the reports can't be written 
# due out-of-diskspace errors. Setting this value to 0 disables the check.
# (Note: this feature is currently not supported under Windows.)
freespace = 64

# Temporary directory containing the files uploaded through Cuckoo interfaces
# (api.py and Django web interface).
tmppath = /tmp

# Path to the unix socket for running root commands.
rooter = /tmp/cuckoo-rooter

[routing]
# Default network routing mode; "none", "internet", or "vpn_name".
# In none mode we don't do any special routing - the VM doesn't have any
# network access (this has been the default actually for quite a while).
# In internet mode by default all the VMs will be routed through the network
# interface configured below (the "dirty line").
# And in VPN mode by default the VMs will be routed through the VPN identified
# by the given name of the VPN (as per vpn.conf).
# Note that just like enabling VPN configuration setting this option to
# anything other than "none" requires one to run utils/rooter.py as root next
# to the Cuckoo instance (as it's required for setting up the routing).
route = none

# Network interface that allows a VM to connect to the entire internet, the
# "dirty line" so to say. Note that, just like with the VPNs, this will allow
# malicious traffic through your network. So think twice before enabling it.
# (For example, to route all VMs through eth0 by default: "internet = eth0").
internet = none

# Routing table name/id for "dirty line" interface. If "dirty line" is
# also default gateway in the system you can leave "main" value. Otherwise add
# new routing table by adding " " line to /etc/iproute2/rt_tables
# (e.g., "200 eth0"). ID and name must be unique across the system (refer to
# /etc/iproute2/rt_tables for existing names and IDs).
rt_table = main

# To route traffic through multiple network interfaces Cuckoo uses 
# Policy Routing with separate routing table for each output interface
# (VPN or "dirty line"). If this option is enabled Cuckoo on start will try 
# to automatically initialise routing tables by copying routing entries from 
# main routing table to the new routing tables. Depending on your network/vpn 
# configuration this might not be sufficient. In such case you would need to 
# initialise routing tables manually. Note that enabling this option won't
# affect main routing table.
auto_rt = yes

[resultserver]
# The Result Server is used to receive in real time the behavioral logs
# produced by the analyzer.
# Specify the IP address of the host. The analysis machines should be able
# to contact the host through such address, so make sure it's valid.
# NOTE: if you set resultserver IP to 0.0.0.0 you have to set the option
# `resultserver_ip` for all your virtual machines in machinery configuration.
ip = 192.168.176.130

# Specify a port number to bind the result server on.
port = 2042

# Maximum size of uploaded files from VM (screenshots, dropped files, log)
# The value is expressed in bytes, by default 10Mb.
upload_max_size = 10485760

[processing]
# Set the maximum size of analyses generated files to process. This is used
# to avoid the processing of big files which may take a lot of processing
# time. The value is expressed in bytes, by default 100Mb.
analysis_size_limit = 104857600

# Enable or disable DNS lookups.
resolve_dns = on

# Enable PCAP sorting, needed for the connection content view in the web interface.
sort_pcap = on

[database]
# Specify the database connection string.
# NOTE: If you are using a custom database (different from sqlite), you have to
# use utf-8 encoding when issuing the SQL database creation statement.
# Examples, see documentation for more:
# sqlite:///foo.db
# postgresql://foo:bar@localhost:5432/mydatabase
# mysql://foo:bar@localhost/mydatabase
# If empty, default is a SQLite in db/cuckoo.db.
connection =

# Database connection timeout in seconds.
# If empty, default is set to 60 seconds.
timeout =

[timeouts]
# Set the default analysis timeout expressed in seconds. This value will be
# used to define after how many seconds the analysis will terminate unless
# otherwise specified at submission.
default = 120

# Set the critical timeout expressed in seconds. After this timeout is hit
# Cuckoo will consider the analysis failed and it will shutdown the machine
# no matter what. When this happens the analysis results will most likely
# be lost. Make sure to have a critical timeout greater than the
# default timeout.
critical = 600

# Maximum time to wait for virtual machine status change. For example when
# shutting down a vm. Default is 60 seconds.
vm_state = 60

**[avd]**
#Path to the local installation of the android emulator
emulator_path = /home/lovina/cuckoo/android-sdk-linux/tools/emulator

#Path to the local installation of the adb - android debug bridge utility.
adb_path = /home/lovina/cuckoo/android-sdk-linux/platform-tools/adb

#Path to the emulator machine files is located
avd_path = /home/lovina/.android/avd

#name of the reference machine that is used to duplicate
reference_machine = aosx

# Specify a comma-separated list of available machines to be used. For each
# specified ID you have to define a dedicated section containing the details
# on the respective machine. (E.g. aosx_1,aosx_2,aosx_3)
#currently supports only 1 machine for network limitations
machines =aosx_1

[aosx_1]
# Specify the label name of the current machine as specified in your
# aosx_1 configuration.
label = aosx_1

# Specify the operating system platform used by current machine
platform = android

# Specify the IP address of the current virtual machine. Make sure that the
# IP address is valid and that the host machine is able to reach it. If not,
# the analysis will fail.
# its always 127.0.0.1 because android emulator networking configurations this the loopback of the host machine
ip = 127.0.0.1

#Specify the port for the emulator as your adb sees it.
emulator_port=5554

# (Optional) Specify the IP of the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here 
# the IP address for the Result Server as your machine sees it. If you don't specify an
# address here, the machine will use the default value from cuckoo.conf.
# NOTE: if you set this option you have to set result server IP to 0.0.0.0 in cuckoo.conf.
# Example:
resultserver_ip = 10.0.2.2

# (Optional) Specify the port for the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here
# the port for the Result Server as your machine sees it. If you don't specify a port
# here, the machine will use the default value from cuckoo.conf.
# Example:
resultserver_port = 2042

**auxiliary conf**

[sniffer]
# Enable or disable the use of an external sniffer (tcpdump) [yes/no].
enabled = yes

# Specify the path to your local installation of tcpdump. Make sure this
# path is correct.
tcpdump = /usr/sbin/tcpdump

# Specify the network interface name on which tcpdump should monitor the
# traffic. Make sure the interface is active.
interface = vboxnet0

# We used to define the network interface to capture on in auxiliary.conf, but
# this has been moved to the "interface" field of each Virtual Machinery
# configuration.

# Specify a Berkeley packet filter to pass to tcpdump.
# bpf = not arp

[mitm]
# Enable man in the middle proxying (mitmdump) [yes/no].
enabled = no

# Specify the path to your local installation of mitmdump. Make sure this
# path is correct.
mitmdump = /usr/local/bin/mitmdump

# Listen port base. Each virtual machine will use its own port to be
# able to make a good distinction between the various running analyses.
# Generally port 50000 should be fine, in this case port 50001, 50002, etc
# will also be used - again, one port per analyses.
port_base = 50000

# Script file to interact with the network traffic. Please refer to the
# documentation of mitmproxy/mitmdump to get an understand of their internal
# workings. (https://mitmproxy.org/doc/scripting/inlinescripts.html)
script = data/mitm.py

# Path to the certificate to be used by mitmdump. This file will be
# automatically generated for you if you run mitmdump once. It's just that
# you have to copy it from ~/.mitmproxy/mitmproxy-ca-cert.p12 to somewhere
# in the analyzer/windows/ directory. Recommended is to write the certificate
# to analyzer/windows/bin/cert.p12, in that case the following option should
# be set to bin/cert.p12.
certificate = bin/cert.p12

[services]
# Provide extra services accessible through the network of the analysis VM
# provided in separate, standalone, Virtual Machines [yes/no].
enabled = no

# Comma-separated list with each Virtual Machine containing said service(s).
services = honeyd

# Time in seconds required to boot these virtual machines. E.g., some services
# will only get online after a minute because initialization takes a while.
timeout = 0

**reporting conf**

# Enable or disable the available reporting modules [on/off].
# If you add a custom reporting module to your Cuckoo setup, you have to add
# a dedicated entry in this file, or it won't be executed.
# You can also add additional options under the section of your module and
# they will be available in your Python class.

[jsondump]
enabled = yes
indent = 4
encoding = latin-1
calls = yes

[reporthtml]
enabled = yes

[mongodb]
enabled = yes
host = 127.0.0.1
port = 27017
db = cuckoo
store_memdump = yes
paginate = 100

[moloch]
enabled = no
# If the Moloch web interface is hosted on a different IP address than the
# Cuckoo Web Interface then you'll want to override the IP address here.
# host = 127.0.0.1
#
# Following are various configurable settings. When in use of a recent version
# of Moloch there is no need to change any of the following settings as they
# represent the defaults.
#
# moloch_capture = /data/moloch/bin/moloch-capture
# conf = /data/moloch/etc/config.ini
# instance = cuckoo

[reportandroidhtml]
enabled = yes

**Comment by doomedraven** [comment]
Dev = rc2

**Comment by lovina37** [comment]
lovina@lovina-Aspire-E1-571:~/Downloads/cuckoo$ python cuckoo.py

  eeee e   e eeee e   e  eeeee eeeee 
  8  8 8   8 8  8 8   8  8  88 8  88 
  8e   8e  8 8e   8eee8e 8   8 8   8 
  88   88  8 88   88   8 8   8 8   8 
  88e8 88ee8 88e8 88   8 8eee8 8eee8

 Cuckoo Sandbox 2.0-dev
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

Oops! Cuckoo failed in an unhandled exception!
Sometimes bugs are already fixed in the development release, it is therefore recommended to retry with the latest development release available https://github.com/cuckoosandbox/cuckoo
If the error persists please open a new issue at https://github.com/cuckoosandbox/cuckoo/issues

=== Exception details ===
Cuckoo version: 2.0-dev
OS version: posix
OS release: Ubuntu 16.04 xenial
Python version: 2.7.12
Machine arch: x86_64
Modules: adium-theme-ubuntu:0.3.4 alembic:0.8.0 backports-abc:0.5 backports.shutil-get-terminal-size:1.0.0 beautifulsoup4:4.4.1 bleach:1.5.0 certifi:2017.1.23 cffi:1.6.0 chardet:2.3.0 cheetah:2.4.4 click:6.7 colorama:0.3.7 configparser:3.5.0 cryptography:1.3.2 decorator:4.0.11 distorm3:3.3.4 django:1.8.4 dnspython:1.12.0 dpkt:1.8.7 ecdsa:0.13 elasticsearch:2.2.0 entrypoints:0.2.2 enum34:1.0.4 et-xmlfile:1.0.1 flask:0.10.1 functools32:3.2.3.post2 html5lib:0.9999999 httpreplay:0.1.17 idna:2.0 ipaddress:1.0.14 ipykernel:4.5.2 ipython-genutils:0.1.0 ipython:5.1.0 ipywidgets:5.2.2 itsdangerous:0.24 jdcal:1.3 jinja2:2.8 jsbeautifier:1.5.10 jsonschema:2.5.1 jupyter-client:4.4.0 jupyter-console:5.0.0 jupyter-core:4.2.1 jupyter:1.0.0 libvirt-python:1.3.1 lxml:3.6.0 m2crypto:0.25.1 mako:1.0.1 markupsafe:0.23 mistune:0.7.3 nbconvert:5.1.1 nbformat:4.2.0 ndg-httpsclient:0.4.0 notebook:4.3.1 oletools:0.42 openpyxl:2.4.1 pandocfilters:1.4.1 paramiko:2.1.1 pathlib2:2.2.1 peepdf:0.3.2 pefile2:1.2.11 pexpect:4.2.1 pickleshare:0.7.4 pillow:3.2.0 pip:9.0.1 prompt-toolkit:1.0.9 ptyprocess:0.5.1 pyasn1:0.1.8 pycparser:2.14 pycrypto:2.6.1 pycurl:7.43.0 pygments:2.2.0 pyliblzma:0.5.3 pymisp:2.4.49 pymongo:3.0.3 pyopenssl:0.15.1 pysqlite:1.0.1 python-dateutil:2.4.2 python-editor:0.3 python-magic:0.4.6 pythonaes:1.0 pyzmq:16.0.2 qtconsole:4.2.1 requests:2.7.0 rpm-python:4.12.0.1 scandir:1.4 setuptools:20.7.0 simplegeneric:0.8.1 singledispatch:3.4.0.3 six:1.9.0 sqlalchemy:1.0.8 terminado:0.6 testpath:0.3 tlslite-ng:0.6.0a3 tornado:4.4.2 traitlets:4.3.1 typing:3.5.3.0 ujson:1.35 unity-lens-photos:1.0 urlgrabber:3.9.1 urllib3:1.20 vboxapi:1.0 volatility:2.6 wakeonlan:0.2.2 wcwidth:0.1.7 werkzeug:0.10.4 wheel:0.29.0 widgetsnbextension:1.2.6 xenapi:1.2 yara-python:3.4.0 yum-metadata-parser:1.1.4

Traceback (most recent call last):
  File "cuckoo.py", line 120, in <module>
    test=args.test)
  File "cuckoo.py", line 70, in cuckoo_init
    init_yara()
  File "/home/lovina/Downloads/cuckoo/lib/cuckoo/core/startup.py", line 270, in init_yara
    with open(index_path, "w") as index_handle:
IOError: [Errno 13] Permission denied: '/home/lovina/Downloads/cuckoo/data/yara/index_binaries.yar'

**Comment by doomedraven** [comment]
you don't need open new issue for the same stuff, and is just permission, use `chown` to change permission and done

**Comment by lovina37** [comment]
okay I will search for it

**Comment by lovina37** [comment]
You do not have the m2crypto library installed preventing certificate extraction: pip install m2crypto

sudo -H pip install m2crypto
[sudo] password for lovina: 
Requirement already satisfied: m2crypto in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied: typing in /usr/local/lib/python2.7/dist-packages (from m2crypto)

why this error then ??

**Comment by doomedraven** [comment]
use search on issues, that was discussed a lot of times ;)

**Comment by lovina37** [comment]
okay :)

**Comment by lovina37** [comment]
@doomedraven 

I solved that using sudo -H pip install M2Crypto==0.24.0 .

Now I am getting 2017-02-13 18:58:52,682 [modules.processing.network] ERROR: Unable to open /home/lovina/Downloads/cuckoo/storage/analyses/3/dump_sorted.pcap

do you know what can be the reason ??

**Comment by doomedraven** [comment]
Did you execute chown command to change permissions recursive for all files?

**Comment by lovina37** [comment]
all files meaning entire cuckoo folder ??

**Comment by lovina37** [comment]
@doomedraven 
I solved it with
chmod -R 0777 cuckoo

Now I get this there exist snapshot in

/home/lovina/vmware/cuckoo2/cuckoo2-Snapshot2.vmsn

![image](https://cloud.githubusercontent.com/assets/14849570/22910301/0a1affde-f227-11e6-8658-2d883e208b30.png)

**Comment by doomedraven** [comment]
Are you sure what file exist?

**Comment by lovina37** [comment]
lovina@lovina-Aspire-E1-571:~/Downloads/cuckoo$ python cuckoo.py

   ______   __  __   ______   ___   ___   ______   ______      
  /_____/\ /_/\/_/\ /_____/\ /___/\/__/\ /_____/\ /_____/\     
  \:::__\/ \:\ \:\ \\:::__\/ \::.\ \\ \ \\:::_ \ \\:::_ \ \    
   \:\ \  __\:\ \:\ \\:\ \  __\:: \/_) \ \\:\ \ \ \\:\ \ \ \   
    \:\ \/_/\\:\ \:\ \\:\ \/_/\\:. __  ( ( \:\ \ \ \\:\ \ \ \  
     \:\_\ \ \\:\_\:\ \\:\_\ \ \\: \ )  \ \ \:\_\ \ \\:\_\ \ \ 
      \_____\/ \_____\/ \_____\/ \__\/\__\/  \_____\/ \_____\/

 Cuckoo Sandbox 2.0-dev
 www.cuckoosandbox.org
 Copyright (c) 2010-2015

 Checking for updates...
 You are running a development version! Current stable is 2.0-rc1.
2017-02-14 18:54:39,083 [root] INFO: Updated running task ID 16 status to failed_analysis
2017-02-14 18:54:39,126 [lib.cuckoo.core.scheduler] INFO: Using "vmware" as machine manager
2017-02-14 18:54:39,855 [lib.cuckoo.core.scheduler] INFO: Loaded 1 machine/s
2017-02-14 18:54:39,881 [lib.cuckoo.core.scheduler] INFO: Waiting for analysis tasks.
2017-02-14 18:54:44,300 [lib.cuckoo.core.scheduler] INFO: Starting analysis of FILE "8fd206a8-f075-11e6-b486-80e65024849a.file" (task #17, options "")
2017-02-14 18:54:44,404 [lib.cuckoo.core.scheduler] INFO: File already exists at "/home/lovina/Downloads/cuckoo/storage/binaries/ec52bc159b12ec20621cc6842a16ba4f726919c48ec2e8882f9b5a34c6710763"
2017-02-14 18:54:44,561 [lib.cuckoo.core.scheduler] INFO: Task #17: acquired machine cuckoo2 (label=/home/lovina/vmware/cuckoo2/cuckoo2.vmx)
2017-02-14 18:54:44,582 [modules.auxiliary.mitm] INFO: Started mitm interception with PID 7708 (ip=192.168.122.1, port=50000).
2017-02-14 18:54:44,590 [modules.auxiliary.sniffer] INFO: Started sniffer with PID 7709 (interface=virbr0, host=172.16.234.128, pcap=/home/lovina/Downloads/cuckoo/storage/analyses/17/dump.pcap)
2017-02-14 18:55:15,133 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=cuckoo2, ip=172.16.234.128)
2017-02-14 19:00:36,584 [lib.cuckoo.core.scheduler] ERROR: Error from the Cuckoo Guest: cuckoo2: the guest initialization hit the critical timeout, analysis aborted.
2017-02-14 19:01:03,348 [modules.processing.behavior] WARNING: Analysis results folder does not exist at path '/home/lovina/Downloads/cuckoo/storage/analyses/17/logs'.
2017-02-14 19:01:05,827 [modules.processing.network] ERROR: Unable to open /home/lovina/Downloads/cuckoo/storage/analyses/17/dump_sorted.pcap
2017-02-14 19:01:08,439 [lib.cuckoo.core.scheduler] INFO: Task #17: reports generation completed (path=/home/lovina/Downloads/cuckoo/storage/analyses/17)
2017-02-14 19:01:08,570 [lib.cuckoo.core.scheduler] INFO: Task #17: analysis procedure completed

**Comment by lovina37** [comment]
It started but it still aborts something like guest initialization error ...

**Comment by doomedraven** [comment]
`2017-02-14 19:00:36,584 [lib.cuckoo.core.scheduler] ERROR: Error from the Cuckoo Guest: cuckoo2: the guest initialization hit the critical timeout, analysis aborted.` looks like agent is not configured correctly

**Comment by lovina37** [comment]
@doomedraven  should I do all the steps again ?? or any particular configuration file that I need to check ??

**Comment by doomedraven** [comment]
check links from this thread, probably it can helps you https://github.com/cuckoosandbox/cuckoo/issues/1054

**Comment by lovina37** [comment]
@doomedraven 
Error: adding task to database

WHY DO U THINK THIS ERROR COMES AFTER EVERYTHING IS PROPERLY SET

**Comment by lovina37** [comment]
@doomedraven 
![image](https://cloud.githubusercontent.com/assets/14849570/23004953/71d262c4-f3c7-11e6-83cf-dbc7f877e60a.png)

see this

**Comment by doomedraven** [comment]
looks like you have problems with vmware, as vmrun generated error

**Comment by lovina37** [comment]
![screenshot from 2017-02-18 20-03-20](https://cloud.githubusercontent.com/assets/14849570/23098187/423a3d92-f616-11e6-9fac-6eac7fb587ea.png)

**Comment by lovina37** [comment]
@doomedraven 

what to do ??

**Comment by lovina37** [comment]
@doomedraven 

![image](https://cloud.githubusercontent.com/assets/14849570/23682428/c92a936c-0361-11e7-982a-3804e6575d9b.png)

**Comment by lovina37** [comment]
@doomedraven 

have u face an issue where u get this : 

Uploaded file length larger than upload_max_size, stopping upload


and this one if yes how did you solve it ??

Uploaded file length larger than upload_max_size, stopping upload

**Comment by doomedraven** [comment]
check this https://github.com/cuckoosandbox/cuckoo/blob/master/conf/cuckoo.conf#L123
for the second one i think is the monitor outdated, not sure

**Comment by lovina37** [comment]
@where is the  monitor stored ??

**Comment by lovina37** [comment]
@doomedraven 

Its since a long time I have not been able to do android apk analysis. I want to get cuckoodroid working. 
Any setup is finr for me. I have tried a lot but every time I get an error.

My request to you is if you have had got cuckoodroid working for any type of setup please can u guide me. 

I complete all the configurations and follow the steps but then I dont know if there is compatibility issue between cuckoo new version and cuckoodroid. 

I get several errors. Please tell me what has to be done Or else suggest me any other dynamic analysis tool for android apk analysis. Please I would be grateful.

**Comment by waqarrashid33** [comment]
@lovina37 Did you made it work?
I had some success with cuckoo 1.2 for android analysis.

**Comment by seantree** [comment]
@waqarrashid33 I have the issues in android analysis.. can you please guide me.
I have installed the centos 7 and inside that I have installed latest cuckoo 2.0.5 and in that I am using virtualbox -->Ubuntu_14_64Bit--> after installing the packages according to the document inside the guest OS I am facing the issue while running the emulator Contacts.apk , superapk are going to be push.
I got following type of error:
Push command not found 

and it's not copying or installing the files in android emulator. I tried every step

**Comment by doomedraven** [comment]
2.0.5 doesnt work for android

**Comment by seantree** [comment]
@doomedraven Thanks for the fast replying, Ok I will go for the Cuckoo 1.2 Version but the thing is right now I am preparing the Guest Machine and it's generating the error at emulator time when I am executing the script of files copying and executing. it's showing push command not found and not copying the apk files.

**Comment by jbremer** [comment]
Currently Android isn't supported in Cuckoo `2.0.0` or later. Going to close this issue. For a working Cuckoo Droid, please check out Idan's "official" patches. Sorry for this, we hope to address these issues this year.
